src/datasets/commands/dummy_data.py

```diff
@@ -49,84 +280,125 @@ def run(self):
             mock_dl_manager = MockDownloadManager(
-                dataset_name=self._dataset_name, config=config, version=version, is_local=True
+                dataset_name=self._dataset_name,
+                config=config,
+                version=version,
+                is_local=True,
+                load_existing_dummy_data=False,
             )
 
-            dummy_data_folder = os.path.join(self._path_to_dataset, mock_dl_manager.dummy_data_folder)
-            logger.info(f"Creating dummy folder structure for {dummy_data_folder}... ")
-            os.makedirs(dummy_data_folder, exist_ok=True)
-
-            try:
-                generator_splits = dataset_builder._split_generators(mock_dl_manager)
-            except FileNotFoundError as e:
-
-                print(
-                    f"Dataset {self._dataset_name} with config {config} seems to already open files in the method `_split_generators(...)`. You might consider to instead only open files in the method `_generate_examples(...)` instead. If this is not possible the dummy data has to be created with less guidance. Make sure you create the file {e.filename}."
+            if self._auto_generate:
+                self._autogenerate_dummy_data(
+                    dataset_builder=dataset_builder,
+                    mock_dl_manager=mock_dl_manager,
+                    keep_uncompressed=self._keep_uncompressed,
                 )
+            else:
+                self._print_dummy_data_instructions(dataset_builder=dataset_builder, mock_dl_manager=mock_dl_manager)
+
+    def _autogenerate_dummy_data(self, dataset_builder, mock_dl_manager, keep_uncompressed):
+        dl_cache_dir = os.path.join(dataset_builder._cache_dir_root, "downloads")
+        download_config = DownloadConfig(cache_dir=dl_cache_dir)
+        dl_manager = DummyDataGeneratorDownloadManager(
+            dataset_name=self._dataset_name, mock_download_manager=mock_dl_manager, download_config=download_config
+        )
+        dataset_builder._split_generators(dl_manager)
+        dl_manager.auto_generate_dummy_data_folder(
+            n_lines=self._n_lines,
+            json_field=self._json_field,
+            xml_tag=self._xml_tag,
+            match_text_files=self._match_text_files,
+        )
+        if not keep_uncompressed:
+            path_do_dataset = os.path.join(mock_dl_manager.datasets_scripts_dir, mock_dl_manager.dataset_name)
+            dl_manager.compress_autogenerated_dummy_data(path_do_dataset)
+        else:
+            generated_dummy_data_dir = os.path.join(self._path_to_dataset, mock_dl_manager.dummy_data_folder)
+            logger.info(
+                f"Dummy data generated in directory '{generated_dummy_data_dir}' but kept uncompressed. "
+                "Please compress this directory into a zip file to use it for dummy data tests."
+            )
+        return dl_manager
 
-            files_to_create = set()
```

Comment:
I moved all this part in the function below without changes.
Sorry github makes it kinda hard to read